docs/Modules/ecdsa/ecdsasmartcontracts.mdx

```diff
@@ -0,0 +1,85 @@
+---
+sidebar_label: "Smart Contract Deep Dive"
+sidebar_position: 2
+description: "In-depth smart contract insights into Biconomy's ECDSA Ownership Registry Module."
+---
+
+# 📜 Smart Contract Deep Dive 🛠️
+
+## Introduction
+
+This document delves into the `EcdsaOwnershipRegistryModule` for Biconomy Smart Accounts, focusing on key functionalities and security aspects.
+
+## Core Functionalities
+
+### User Operation Validation (`validateUserOp`) 🛡️
+
+```solidity
+// Validates user operations signed by an EOA
+function validateUserOp(UserOperation calldata userOp, bytes32 userOpHash)
+    external view returns (uint256) {
+    (bytes memory cleanEcdsaSignature, ) = abi.decode(userOp.signature, (bytes, address));
+    if (_verifySignature(userOpHash, cleanEcdsaSignature, userOp.sender)) {
+        return VALIDATION_SUCCESS;
+    }
+    return SIG_VALIDATION_FAILED;
+}
+
+```
+
+- **Objective**: Ensures the authenticity of user operations.
+- **Method**: Decodes and verifies the user operation's signature.
```

Comment:
COuld be great adding details, why we take a part of the `userOp.signature` only